woocommerce/woocommerce-ios
# [Free trial survey] Show survey upon tapping local notification

Part of: #10266

## Description
Displays the survey screen upon tapping local notification. 

Changes
- Show survey screen upon tapping survey local notification.
- Unit tests for view model and coordinator. 
- Track events upon showing and submitting survey. 

## Testing instructions
- Build and run the app on a physical device for easy testing.
- Create a new store with a free trial subscription.
- Open the device Settings app to switch to 23h58m after the time of the store creation.
- After 2 minutes, a notification asking for a survey should be displayed.
- Tap the notification, the app should be opened. 
- Observe that the free trial survey screen is presented and the following event is tracked.
```
🔵 Tracked free_trial_survey_displayed, properties: [AnyHashable("blog_id"): 12345, AnyHashable("is_wpcom_store"): true, AnyHashable("source"): "free_trial_survey_24h_after_free_trial_subscribed"]
```
- Selecting an option from the survey and submitting the survey should track the following event
```
🔵 Tracked free_trial_survey_sent, properties: [AnyHashable("source"): "free_trial_survey_24h_after_free_trial_subscribed", AnyHashable("blog_id"): 123456, AnyHashable("survey_option"): "<option>", AnyHashable("is_wpcom_store"): true, AnyHashable("free_text"): "<TextEntered>"]
```

## Screenshots

<img src="https://github.com/woocommerce/woocommerce-ios/assets/524475/f93926ff-487d-4f83-b5f4-2117da8e09b8" width="320"/>

https://github.com/woocommerce/woocommerce-ios/assets/524475/5956e991-b756-4258-9829-a95d4ca4a098


---
